### PR TITLE
[bugfix] Fix Slurm backend failures when trying node lists with inexistent nodes

### LINKED ISSUE #694: Improper handling of showing nodes by name via scontrol
When executing `scontrol show -a -o node <nodenames>` the output may contain lines such as: `Node foo not found`  while other lines contain the proper node information. In this case, although the `scontrol` return status is **!=0**, the output should be parsed ignoring the nodes that are not found.

This has happened a few times when the CI is running the unittests. Take a look [here](https://jenkins.cscs.ch/view/Reframe%20Regression/job/ReframeCI/1565/).

### PR DESCRIPTION
**Pull request by kraushm**
Modifying the way the set of available nodes is built in SLURM, in order to prevent erroneous entries (e. g. "not found")

Fixes #694

### CODE REVIEW

**Comment by vkarak** [comment]
@kraushm Why this is a WIP?

**Comment by kraushm** [comment]
@jenkins-cscs retry kesch

File: reframe/core/schedulers/slurm.py [hunk 1]
```diff
@@ -240,14 +240,17 @@ def _get_reservation_nodes(self, reservation):
...
 
     def _get_nodes_by_name(self, nodespec):
-        try:
-            completed = _run_strict('scontrol -a show -o node %s' % nodespec)
-        except SpawnedProcessError as e:
-            raise JobError('could not retrieve the node description '
-                           'of nodes: %s' % nodespec) from e
-
+        completed = os_ext.run_command('scontrol -a show -o node %s'
+                                       % nodespec)
```
Comment (teojgo) [inline reframe/core/schedulers/slurm.py]: split line after the operator `%`

**Comment by kraushm** [comment]
@jenkins-cscs retry daint

**Pull request merged**